pkg/serviceaccounts/controllers/docker_registry_service.go

```diff
@@ -168,20 +255,24 @@ func (e *DockerRegistryServiceController) handleLocationChange(serviceLocation s
 
 		dockercfgMap := map[string]credentialprovider.DockerConfigEntry(*dockercfg)
 		keys := sets.KeySet(reflect.ValueOf(dockercfgMap))
-		if len(keys) != 1 {
+		if len(keys) == 0 {
```

Comment:
> in this case, or in the case where an update below fails on a conflict, remind me how the secret eventually gets cleaned up or fixed? is it on the dockercfgController resync?

Looks like an existing problem.  `resync` will call a series of `Updates`, those updates won't take any action because they look like dupes.  

Simplest approach? Retry twice, then after that simply indicate a failure needing a full reset.  Fire off a go func to rekick the job.  That gives some tolerance.